code/game/machinery/roulette_machine.dm

```diff
@@ -0,0 +1,383 @@
+/obj/machinery/roulette
+	name = "Roulette Table"
+	desc = "A computerized roulette table."
+	icon = 'icons/obj/machines/roulette.dmi'
+	icon_state = "idle"
+	density = TRUE
+	use_power = IDLE_POWER_USE
+	anchored = FALSE
+	idle_power_usage = 10
+	active_power_usage = 100
+	max_integrity = 500
+	armor = list("melee" = 45, "bullet" = 30, "laser" = 30, "energy" = 30, "bomb" = 10, "bio" = 30, "rad" = 30, "fire" = 30, "acid" = 30)
+	var/static/list/numbers = list("0" = "green", "1" = "red", "3" = "red", "5" = "red", "7" = "red", "9" = "red", "12" = "red", "14" = "red", "16" = "red",\
+	"18" = "red", "19" = "red", "21" = "red", "23" = "red", "25" = "red", "27" = "red", "30" = "red", "32" = "red", "34" = "red", "36" = "red",\
+	"2" = "black", "4" = "black", "6" = "black", "8" = "black", "10" = "black", "11" = "black", "13" = "black", "15" = "black", "17" = "black", "20" = "black",\
+	"22" = "black", "24" = "black", "26" = "black", "28" = "black", "29" = "black", "31" = "black", "33" = "black", "35" = "black")
+
+	var/chosen_bet_amount = 10
+	var/house_balance = 3500 //placeholder
+	var/account_balance = 100 //placeholder
+	var/chosen_bet_type = 0
+	var/last_anti_spam = 0
+	var/anti_spam_cooldown = 20
+	var/obj/item/card/id/my_card
+	var/playing = FALSE
+	var/locked = FALSE
+	var/drop_dir = SOUTH
+	var/list/coin_values = list(/obj/item/coin/diamond = 100, /obj/item/coin/gold = 25, /obj/item/coin/silver = 10, /obj/item/coin/iron = 1) //Make sure this is ordered from left to right.
+	var/list/coins_to_dispense = list()
+	var/datum/looping_sound/jackpot/jackpot_loop
+
+/obj/machinery/roulette/Initialize()
+	. = ..()
+	jackpot_loop = new(list(src), FALSE)
+	wires = new /datum/wires/roulette
+
+/obj/machinery/roulette/obj_break(damage_flag)
+	prize_theft(0.05)
+	. = ..()
+
+/obj/machinery/roulette/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	if(stat & MAINT)
+		return
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		var/datum/asset/spritesheet/simple/assets = get_asset_datum(/datum/asset/spritesheet/simple/roulette)
+		assets.send(user)
+		ui = new(user, src, ui_key, "roulette", name, 455, 520, master_ui, state)
+		ui.open()
+
+/obj/machinery/roulette/ui_data(mob/user)
+	var/list/data = list()
+	data["IsAnchored"] = anchored
+	data["BetAmount"] = chosen_bet_amount
+	data["BetType"] = chosen_bet_type
+	data["HouseBalance"] = my_card?.registered_account.account_balance
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/obj/item/card/id/C = H.get_idcard(TRUE)
+		if(C)
+			data["AccountBalance"] = C.registered_account.account_balance
+		else
+			data["AccountBalance"] = 0
+		data["CanUnbolt"] = (H.get_idcard() == my_card)
+
+	var/datum/asset/spritesheet/simple/assets = get_asset_datum(/datum/asset/spritesheet/simple/roulette)

```

Comment:
You don't really need to be sending these every time if the spritesheet is static.